gossip/src/duplicate_shred_handler.rs

```diff
@@ -85,7 +97,12 @@ impl DuplicateShredHandler {
         self.last_root = last_root;
         if let Ok(bank_fork) = self.bank_forks.try_read() {
             let root_bank = bank_fork.root_bank();
+            self.cached_epoch_schedule = Some(root_bank.epoch_schedule().clone());
             let epoch_info = root_bank.get_epoch_info();
+            self.enable_gossip_duplicate_proof_ingestion_epoch = root_bank
+                .feature_set
+                .activated_slot(&feature_set::enable_gossip_duplicate_proof_ingestion::id())
+                .map(|slot| root_bank.epoch_schedule().get_epoch(slot));
```

Comment:
nit: Seems like this could be gated behind a `if self.enable_gossip_duplicate_proof_ingestion_epoch.is_none()` check